server/src/main/java/org/elasticsearch/index/mapper/IdFieldMapper.java

```diff
@@ -156,11 +165,6 @@ public Query termsQuery(List<?> values, QueryShardContext context) {
                     CircuitBreakerService breakerService,
                     MapperService mapperService
                 ) {
-                    if (mapperService.isIdFieldDataEnabled() == false) {
-                        throw new IllegalArgumentException("Fielddata access on the _id field is disallowed, "
-                            + "you can re-enable it by updating the dynamic cluster setting: "
-                            + IndicesService.INDICES_ID_FIELD_DATA_ENABLED_SETTING.getKey());
-                    }
                     deprecationLogger.deprecate("id_field_data", ID_FIELD_DATA_DEPRECATION_MESSAGE);
```

Comment:
I kept the deprecation message here so we don't issue a warning on every call to field caps.